ipatests/test_integration/test_replica_promotion.py

```diff
@@ -887,12 +892,29 @@ def _check_config(self, enabled=(), hidden=()):
             assert values.get(hservice, set()) == hidden
 
     def test_hidden_replica_install(self):
-        # TODO: check that all services are running on hidden replica
         self._check_server_role(self.master, 'enabled')
         self._check_server_role(self.replicas[0], 'hidden')
         self._check_dnsrecords([self.master], [self.replicas[0]])
         self._check_config([self.master], [self.replicas[0]])
 
+    def test_ipahealthcheck_hidden_replica(self):
+        """Ensure that ipa-healthcheck runs successfully on all members
+        of an IPA cluster that includes a hidden replica.
+        """
+        # verify state
+        self._check_config([self.master], [self.replicas[0]])
+        # A DNA range is needed on the replica for ipa-healthcheck to work.
+        # Create a user so that the replica gets a range.
+        tasks.user_add(self.replicas[0], 'testuser')
+        tasks.user_del(self.replicas[0], 'testuser')
+        for srv in (self.master, self.replicas[0]):
+            tasks.install_packages(srv, HEALTHCHECK_PKG)
```

Comment:
Can we move this to test class level so if we add more tests in future, this pkg install is applied for those tests as well?